### ISSUE TITLE
[FEATURE] Show signed commits in GitQlient (when "showSignature = true" is setup)

### ISSUE THREAD

**Issue by band-a-prend** [francescmm/GitQlient#161]
**Describe the bug**
I use global git config option `showSignature = true` and then I try to open any cloned git repo with GitQlient
then any information isn't displayed in application window:
![showSignature_true](https://user-images.githubusercontent.com/18756734/95657302-71225f80-0b1c-11eb-89c1-d34f5fb6a45e.png)

If I change local git config to `showSignature = false` then information about commits is displayed:
![show_signature_false](https://user-images.githubusercontent.com/18756734/95657340-b777be80-0b1c-11eb-8017-61e94f16eb3e.png)

**To Reproduce**
1. Build GitQlient (e.g. from master)
2. Go to any directory with cloned git repo (e.g. https://github.com/francescmm/GitQlient)
and setup for for global or local git config an aption
```
[log]
	showSignature = true
```
3. Run GitQlient and open repo with signed commits that was setup in previous step .
4. The empty GitQlient application winows is displayed

**Expected behavior**
The GitClient application window with commits and its signatures are expected to see.

**Desktop (please complete the following information):**
 - OS: Gentoo Linux
 - Release version or SHA: https://github.com/francescmm/GitQlient/commit/d61ed0b0f7a62a9525bbe648a861d5b6e8c6a879
 - If manually compiled, compiler version: 

**Additional context**
Some over application have various issues with `showSignature = true`. The `lazygit` early failed in this case - already resolved issue. But `qgit` is still affected to fail with this option being setup.

**Comment by francescmm** [comment]
Hi @band-a-prend Thanks for the request. Unfortunately right now GitQlient doesn't support `showSignature` active.

It seams fairly easy to at least support that, although for any other options of signature I would need a Feature Request.

I'll mark this as a feature request and see if I can release GitQlient 1.3.0 with it.

**Comment by band-a-prend** [comment]
I hope this feature could be implemented to work with showSignature option and display signature. Some git repositories reject pushing non-signed commits. So it's convenient to see if commit signed or not.

Although as workaround could be setup globally always sign commits and then if it will sign on rebase too then no reason to worry about signature.

**Comment by francescmm** [comment]
I'm sorry but I was feeling bored and I've implemented it :rofl: 

Something like this it's okay? :yum: 

![image](https://user-images.githubusercontent.com/4884514/95661203-d6328100-0b2d-11eb-8f10-d280ecd74dc8.png)

The information of the signature is also present in the _tooltip_. I might need to work with the performance though. There are a lot of string operations happening that are definitely slowing down the refresh of the view.

**Comment by francescmm** [comment]
For now is present in _develop_ branch, so feel free to checkout and compile if you wanted. Develop is not as stable as master, but it shouldn't give you any problems (is what I use in my day-to-day at work).

**Comment by band-a-prend** [comment]
It's really quickly implemented! Great thanks, now it shows commits tree.

The only thing the displayed information is reverted depend on `showSignature` git config value.
With `showSignature = false` the "check" icon is displayed with info "Commit signed!" and "GPG key log size ..."
![crop_new_showSignature_false](https://user-images.githubusercontent.com/18756734/95672678-9f9b4b80-0bab-11eb-8c40-8927f678d11c.png)

While with `showSignature = true` git config value there is no additional icon shown and displayed info
![crop_new_showSignature_true](https://user-images.githubusercontent.com/18756734/95672698-bb065680-0bab-11eb-8a82-775475c3c70e.png)

**Comment by francescmm** [comment]
Thanks for noticing that! Definitely, it was too late yesterday to have all the work done. I just pushed a commit that will solve that as well as unifying how the information is processed.

Now, instead of being CommitInfo class who processes all the information, is the GitRepoLoader class.

Another thing I've discovered is that the performance penalty I get comes from _git_ it self. It takes a longer time to return the log with the GPG signature (and it adds trice the log lines to process). I've decreased the processing time to barely the same time as before the signature was added, and also improve the UX for it, but Git it's the culprit on this.

**Comment by band-a-prend** [comment]
Yes, performance penalty from git command is problem. I can't open https://github.com/gentoo/gentoo cloned repo. Maybe because my old pc system too. I see git load cpu in htop for a long time while trying to show repo in gitqlient. But after several minutes it still goes on.

**Comment by francescmm** [comment]
Another thing I have to do is to enable the "limit" of commits to load (right now "--all").

In addition, there is another instruction that is taking a long time and is the one I use to calculate the distances between the local/origin/master for every local branch.

I just clone the _gentoo_ repo and apart of the eternal cloning, the rest went through quite smoothly so probably the above improvements can help your PC :smiley: 

I'll probably do them in a different issue, for now I'll close this as done.

**Comment by band-a-prend** [comment]
Thank you for resolving issue.

> improvements can help your PC

I think it's generally my old PC issue now. The command that displayed in htop process list is
```
git log --date-order --no-color --log-size --parents --boundary -z --pretty=format:%m%HX%P%n%cn<%ce>%n%an<%ae>%n%at%n%s%n%b --all
```
When I tried to run it's inside repository then I receive errors about unknown  `<%ce>` and `<%ae>`. So I ran 
```
git log --date-order --no-color --log-size --parents --boundary -z --pretty=format:%m%HX%P%n%cn%ce%n%an%ae%n%at%n%s%n%b --all
```
It run fast but the output seems is through `more` or `less`. So it I add to the commad ` | wc -l` then ... I need wait about 24 minutes to get number of output lines that is more then 4 million strings (~ 383900 commits). 
For repositories with lesser quantity if commits GitQlient now shows commits list quicker.
